esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -560,6 +561,30 @@ class CEsdlSDSStore : implements IEsdlStore, public CInterface
         return configureLogTransform(bindingId, logTransformName, configTree, overwrite, message);
     }
 
+    void preserveBindingContent(IPTree* src, IPTree* dst, IPTree* exclusion) const
```

Comment:
This behavior is more generic than the name implies.  It's really a sort of mergePTreeExcludingNode() and we could find use for it or evolve it over time if it was named more generically.

Reply:
It is and it isn't. The implementation really assumes that the destination node contains no children and that its attributes take precedence over conflicting values in the source. More than a mergePTreeExcludingNode, its appendChildrenExcludingNode.

I think a generic merge would need to apply a predicate on each source node to decide if it should be included in the merge, and possibly a second predicate to decide how each attribute should be handled.

Reply:
If you define "merge" based on behavior of the existing common function "mergePTree" then this would be mergePTreeExcludingNode().  mergePTree keeps things simple as does this.

The other complex method you described would be something else and probably too complex to include in jptree.hpp  This function shouldn't go there as is either but is close to the keep it simple philosophy.